barretenberg/cpp/src/barretenberg/dsl/acir_format/civc_recursion_constraints.test.cpp

```diff
@@ -0,0 +1,155 @@
+#include "barretenberg/client_ivc/mock_circuit_producer.hpp"
+#include "barretenberg/dsl/acir_format/acir_format.hpp"
+#include "barretenberg/dsl/acir_format/acir_format_mocks.hpp"
+#include "barretenberg/dsl/acir_format/proof_surgeon.hpp"
+#include "barretenberg/stdlib/client_ivc_verifier/client_ivc_recursive_verifier.hpp"
+
+#include <gtest/gtest.h>
+
+using namespace acir_format;
+using namespace bb;
+using namespace bb::stdlib::recursion::honk;
+
+class CivcRecursionConstraintTest : public ::testing::Test {
+  public:
+    using Builder = UltraCircuitBuilder;
+
+    // Types for ClientIVC recursive verifier
+    using Flavor = UltraRollupFlavor;
+    using DeciderProvingKey = DeciderProvingKey_<Flavor>;
+    using VerificationKey = Flavor::VerificationKey;
+    using ClientIVCRecursiveVerifier = stdlib::recursion::honk::ClientIVCRecursiveVerifier;
+
+    // Types for ClientIVC
+    using DeciderZKProvingKey = DeciderProvingKey_<MegaZKFlavor>;
+    using MegaZKVerificationKey = MegaZKFlavor::VerificationKey;
+
+    // Public inputs added by bb to a ClientIVC proof
+    static constexpr size_t PUBLIC_INPUTS_SIZE = bb::HidingKernelIO::PUBLIC_INPUTS_SIZE;
+
+    struct ClientIVCData {
+        std::shared_ptr<MegaZKVerificationKey> mega_vk;
+        ClientIVC::Proof proof;
+    };
+
+    static ClientIVCData get_civc_data(TraceSettings trace_settings)
```

Comment:
This may also need to be updated, depending on when things merge. If so you can look to the CIVC tests to see the relevant new syntax.